app/src/main/java/com/nextcloud/talk/controllers/ChatController.kt

```diff
@@ -577,28 +579,46 @@ class ChatController(args: Bundle) : BaseController(args), MessagesListAdapter
         }
     }
 
-    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+    override fun onActivityResult(requestCode: Int, resultCode: Int, intent: Intent?) {
         if (requestCode == REQUEST_CODE_CHOOSE_FILE) {
             if (resultCode == RESULT_OK) {
-                uploadFile(data)
+                checkNotNull(intent)
+                val files: MutableList<String> = ArrayList()
+                intent.clipData?.let {
+                    for (index in 0 until it.itemCount) {
+                        files.add(it.getItemAt(index).uri.toString())
+                    }
+                } ?: run {
+                    checkNotNull(intent.data)
+                    intent.data.let {
+                        files.add(intent.data.toString())
+                    }
+                }
+
+                var filenamesWithLinebreaks = "\n"
+                files.forEach {
+                    var filename = UriUtils.getFileName(Uri.parse(it), context)
+                    filenamesWithLinebreaks += filename + "\n"
+                }
+
+                LovelyStandardDialog(activity)
+                        .setPositiveButtonColorRes(R.color.nc_darkGreen)
+                        .setTitle(context?.resources?.getString(R.string.nc_upload_confirm_send)?.let { String.format(it,
```

Comment:
This uses "send these files to" as plural also with one file.

Reply:
done